src/components/input/input_text.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import classNamesDedupe from 'classnames/dedupe';
+
+const INPUT_TEXT_CLASSNAME = 'input-text';
+const INPUT_TEXT_ERROR_CLASSNAME = 'input-text--error';
+
+interface InputTextProps extends React.InputHTMLAttributes<HTMLInputElement> {
+    hasError?: boolean;
+    // prevent creating radio buttons and checkboxes
+    type?: 'text' | 'password' | 'email' | 'url' | 'number';
+}
+
+const InputText = React.forwardRef(
+    (props: InputTextProps, ref: React.Ref<HTMLInputElement>) => {
```

Comment:
I think it is fine to keep the possibility to use `ref` here in case the use of `props` is not appropriate; do you have an idea if there's any case in our applications using `ref` and ref-forwarding?

Reply:
I think this is the most React way to create highly reusable components. Instead of dealing with one use case inside a component (for example setting focus after rendering), it's left up to the consuming component to do (_or not do_) something with a `ref` that's forwarded to the child component.

Either way, it's trivial to add them like this for simple components like this that output a DOM element. It would not make sense to add them to more complex components that output a much larger DOM fragment.